File: spark-java/spark-dataframe/dataframe-aggregation/src/main/java/com/mahfooz/df/aggregation/basic/SparkBasicAggregation.java
package com.mahfooz.df.aggregation.basic;

import org.apache.spark.sql.Dataset;
import org.apache.spark.sql.Row;
import org.apache.spark.sql.SparkSession;
import static org.apache.spark.sql.functions.count;
import static org.apache.spark.sql.functions.countDistinct;

public class SparkBasicAggregation {

    public static void main(String[] args) {

    SparkSession spark = SparkSession
      .builder()
      .master("local[*]")
      .appName("SparkBasicAggregation")
      .getOrCreate();

    String dataHome = System.getenv("DATA_HOME");

    Dataset<Row> df = spark.read()
      .option("header", "true")
      .option("inferSchema", "true")
      .csv(dataHome+"\\FileData\\Csv\\Retail\\all\\online-retail-dataset.csv");

    df.select(count("StockCode")).show();

    df.select(countDistinct("StockCode")).show();

  }

}

File: spark-scala/spark-dataframe/dataframe-aggregation/src/main/scala/com/mahfooz/datafram/aggregation/basic/SparkBasicAggregation.scala
package com.mahfooz.datafram.aggregation.basic

import org.apache.spark.sql.SparkSession
import org.apache.spark.sql.functions.{count, countDistinct}

object SparkBasicAggregation {

  def main(args: Array[String]): Unit = {

    val spark = SparkSession
      .builder()
      .master("local[*]")
      .appName("SparkBasicAggregation")
      .getOrCreate()

    val df = spark.read
      .option("header", "true")
      .option("inferSchema", "true")
      .csv("D:\\data\\retail-data\\all\\online-retail-dataset.csv")

    df.select(count("StockCode")).show()

    df.select(countDistinct("StockCode")).show()

  }

}
